index.js

```diff
@@ -14,6 +21,30 @@ module.exports = (robot, _, Settings = require('./lib/settings')) => {
 
     if (defaultBranch && settingsModified) {
       return Settings.sync(context.github, context.repo(), config)
+        .then(() => {
+          checkOptions.conclusion = 'success'
+        }).catch(res => {
+          checkOptions.conclusion = 'failure'
+          const summary = `
+There was an error while updating the repository settings.
+
+<details><summary>Failed response</summary>
+  <pre>
+  ${JSON.stringify(JSON.parse(res.message), null, 2)}
+  </pre>
+</details>
+`
+          checkOptions.output = {
+            title: 'Settings Probot',
+            summary: summary
+          }
+        }).then(() => {
+          context.github.checks.create(checkOptions)
```

Comment:
Would it be better to create the check and set it to pending status, and then update it after the settings are synced?

The settings sync can take a lot of time to complete: For example, if they had 50 labels it would take around a minute for the sync to finish but the status check wouldn't exist on their commit until the sync finished.